windows-gaming-driver/src/config.rs

```diff
@@ -3,20 +3,124 @@ use std::fs::{File, OpenOptions};
 use std::io::{Read, Write};
 
 use toml;
+use serde_yaml;
 
 #[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq, Copy)]
 pub struct DeviceId {
     pub vendor: u16,
     pub product: u16,
 }
 
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub enum UsbBinding {
+    ById(DeviceId),
+    ByPort { bus: u16, port: u16 },
```

Comment:
I know that and I don't care. It's fine like this. We never need to pass this around.